### PR TITLE
Add missing host and port to Next.js custom server

### PR DESCRIPTION
**Pull request by leoortizz**
### Description

- Pass host and port to Next.js custom server from hosting emulator info. 
- Fallback to `DEFAULT_HOST` and `DEFAULT_PORTS.hosting` since the `serve` command is not passing the emulator info to `prepareFrameworks`.
- Add a TODO to do pass emulator info to `prepareFrameworks` from `serve`.
- Add `hostingEmulatorInfo` parameter to the `Framework`'s `getDevModeHandle` type

### Scenarios Tested

A redirect in a Next.js Middleware should redirect to the appropriate route now, before it was erroring because the host and port were `undefined`.

### CODE REVIEW

File: src/frameworks/next/index.ts [hunk 1]
```diff
@@ -303,17 +303,17 @@ export async function ɵcodegenFunctionsDirectory(sourceDir: string, destDir: st
  * Create a dev server.
  */
 export async function getDevModeHandle(
-  dir: Parameters<NonNullable<Framework['getDevModeHandle']>>[0],
-  hostingEmulatorInfo: Parameters<NonNullable<Framework['getDevModeHandle']>>[1]
+  dir: Parameters<NonNullable<Framework["getDevModeHandle"]>>[0],
```
Comment (jamesdaniels) [inline src/frameworks/next/index.ts:306]: I think we can simplify the typings here a little bit, take on a TODO to namespace it

File: src/serve/index.ts [hunk 2]
```diff
@@ -24,10 +24,10 @@ export async function serve(options: any): Promise<void> {
   options.port = parseInt(options.port, 10);
   if (targetNames.includes("hosting") && config.extract(options).some((it: any) => it.source)) {
     experiments.assertEnabled("webframeworks", "emulate a web framework");
-    
+
     // TODO: pass emulator info as the fourth parameter so that the dev server
     //       can use host and port
-    await prepareFrameworks(targetNames, options, options); 
+    await prepareFrameworks(targetNames, options, options);
```
Comment (jamesdaniels) [inline src/serve/index.ts:30]: Could we pass something in the shape of an emulatorInfo here for now? e.g, `[{name: 'hosting', port: options.port, ... }]`

**Review by jamesdaniels — CHANGES_REQUESTED** [review]
I started a PR to add hostname and port to the custom server in prod, mind taking a peek? We can cut a canary build to test.

https://github.com/FirebaseExtended/firebase-framework-tools/pull/50

**Comment by jamesdaniels** [comment]
Also let's add a line to the changelog

File: src/frameworks/next/index.ts (hunk 1 above)
Comment (leoortizz) [reply src/frameworks/next/index.ts]: Indeed. I replaced it with the explicit types required and added a TODO to reuse it from the `Frameworks` interface

File: src/serve/index.ts (hunk 2 above)
Comment (leoortizz) [reply src/serve/index.ts]: I harcoded it and changed the TODO to get the "real emulator info"

**Pull request merged**